train.py

```diff
@@ -210,6 +222,9 @@ def loss_fn(pred, labels):
             "All the substages will be initialized with random weights with same RNG state which can affect convergence."
         )
 
+    if not checkpoint_loaded and parallel_dims.dp_type == "hsdp":
```

Comment:
i'm not sure what this case is for?

Reply:
We need to synchronize the buffers. That's what DDP performs to ensure that all buffers are the same across ranks. cc., @awgu

Reply:
let's add comments for this